### DISCUSSION TITLE
Growth patterns

### DISCUSSION THREAD

**Discussion by okenk** [pfmc-assessments/canary_2023#21] [Ideas]
I have been fitting growth curves to the combined WCGBTS + Triennial data. Note there are essentially no samples after 2014 (i.e., the last assessment) because those are coming later.

Since this is exploratory, I fit a curve that allowed for variation only in L.inf. I did this for all samples, males only, and females only, with a dividing line at Coos Bay, OR. (Brooks thesis found the split was somewhere between the ports of Coos Bay and Newport.) I found a small but statistically significant difference of ~0.4-0.75 cm. For reference, if you do the same exercise for males and females instead of North and South, you get a difference of 2.6 cm. 

I also played around with the dividing line. If you move it north to the OR-WA border, p-values went up quite a bit, and the N/S term in the female model is not statistically significant (p = 0.15). I moved it down to the CA border, and much less change in p-values.

If you lump males and females together, the AICs are (in order):
OR-CA: 43201
Coos Bay: 43219
OR-WA: 43235
Coastwide: 43246

A plot of the OR-CA split:
![image](https://user-images.githubusercontent.com/205013/200087376-7ba5de4a-89c2-4483-afe8-15c8be5d8c48.png)

I think a split in the middle of Oregon sounds very difficult to implement, and a CA-OR split is well-supported. One challenge will likely be that there aren't nearly as many ages in CA, especially older ages. (Maybe that will change with new age data.):
![image](https://user-images.githubusercontent.com/205013/200083930-ea09296d-cd8d-4a9f-9eba-ef8829c733e2.png)

**Comment by brianlangseth-NOAA** [comment]
This is really interesting, especially because the information from the Brooks thesis was from the survey. Keller et al, and then later Gertseva certainly suggested more pronounced changes. 

What do you mean when you say you did "only variation in Linf"? Does this mean you have sex-specific estimates for Linf yet estimates for K that are combined across sex?

**Comment by okenk** [comment]
Yes, your understanding is correct. However, I figured out how to add factors in nls instead of having to code my own dummy variables, which made it easier to have a separate Linf, k, and t0 for north and south areas. (Residual sigma is still shared.) New AICs:
Coos Bay split: 43149
CA split: 43186
WA split: 43191
coastwide: 43246
coastwide by sex (for comparison sake): 41933 
(Note coastwide by sex actually has 3 more parameters because there is an unsexed category)

So now the middle of OR is most supported, but still seems hard to implement for marginal gains. The difference in the growth curves is subtle. Here is the OR-CA border fitted curves:
![image](https://user-images.githubusercontent.com/205013/200650402-514c43e5-725c-40e1-89b4-42c3b1e90e95.png)

And the Coos Bay split fitted curves:
![image](https://user-images.githubusercontent.com/205013/200650493-1d4864ea-d758-4185-8279-37b833589de9.png)

**Reply by brianlangseth-NOAA** [reply]
I agree Kiva. The differences are pretty small especially given the fewer samples to inform Linf from CA compared to OR-WA. Are we talking differences of Linf still on the order of <1 cm? Differences in K cant be much more than 0.01-0.02 either. 

This is good work. I think we need to revisit Brooks thesis and see what differences they obtained. My recollection was that those differences were much more obvious.

**Comment by okenk** [comment]
The Linf is 1.8 cm different for the Coos Bay split or 0.9 cm for the CA split. Yeah, the k's are ~0.02 apart.

I think part of the issue with the CA split is you end up with a lot fewer samples in the south. If you split at CA the southern Linf SE is 3.3x bigger than the northern Linf SE. If you split at Coos Bay the southern Linf SE is only 2.4x bigger than the northern Linf SE. (Presumably this is due to sample size, especially of old fish.)

If you split at Coos Bay here is the model summary:

``` 
Formula: Length_cm ~ linf[is_south_cb] * (1 - exp(-k[is_south_cb] * (Age_years - 
    t0[is_south_cb])))

Parameters:
       Estimate Std. Error t value Pr(>|t|)    
linf1 53.853955   0.120762 445.953   <2e-16 ***
linf2 55.639922   0.296758 187.492   <2e-16 ***
k1     0.173777   0.001787  97.242   <2e-16 ***
k2     0.149823   0.002478  60.457   <2e-16 ***
t01   -0.364612   0.036933  -9.872   <2e-16 ***
t02   -0.641197   0.045499 -14.093   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 3.136 on 8414 degrees of freedom

Number of iterations to convergence: 6 
```

And here if you split at the California border:
```
Formula: Length_cm ~ linf[is_south_ca] * (1 - exp(-k[is_south_ca] * (Age_years - 
    t0[is_south_ca])))

Parameters:
       Estimate Std. Error t value Pr(>|t|)    
linf1 54.088889   0.116313  465.03   <2e-16 ***
linf2 54.954039   0.380704  144.35   <2e-16 ***
k1     0.170555   0.001566  108.92   <2e-16 ***
k2     0.152238   0.003321   45.84   <2e-16 ***
t01   -0.387063   0.030641  -12.63   <2e-16 ***
t02   -0.685147   0.064839  -10.57   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 3.143 on 8414 degrees of freedom

Number of iterations to convergence: 6 
```

**Comment by brianlangseth-NOAA** [comment]
@okenk Could you update your scripts with your new growth curves (the ones that have all three variables changing)?

**Comment by okenk** [comment]
I just pushed this. (The reproducibility is not dealt with.)

**Comment by okenk** [comment]
Growth curves for coastwide WCGBTS data using additive normal errors (this had lower AIC than multiplicative lognormal):
```
Formula: Length_cm ~ la1[Sex] + (la2[Sex] - la1[Sex]) * (1 - exp(-k[Sex] * 
    (Age - 1)))/(1 - exp(-k[Sex] * 24))

Parameters:
      Estimate Std. Error t value Pr(>|t|)    
la11 11.382147   0.164262   69.29   <2e-16 ***
la12 11.365088   0.147812   76.89   <2e-16 ***
la21 56.385829   0.134634  418.81   <2e-16 ***
la22 50.728545   0.099313  510.80   <2e-16 ***
k1    0.142700   0.001589   89.79   <2e-16 ***
k2    0.175495   0.001670  105.10   <2e-16 ***
---
```
1 is females, 2 is males.

Code to run it is [here](https://github.com/okenk/Assessment_Class/blob/main/code/W03_growth.R).

**Reply by okenk** [reply]
a1 is 1, a2 is 25

**Reply by brianlangseth-NOAA** [reply]
Do these relate to linf via the script right above this one: [here](https://github.com/okenk/Assessment_Class/blob/c50f538b69edf964d3928f3cd076d82bd000d4ed/code/W03_growth.R#L39-L46)

**Reply by okenk** [reply]
No, from [here](https://github.com/okenk/Assessment_Class/blob/c50f538b69edf964d3928f3cd076d82bd000d4ed/code/W03_growth.R#L59-L67).

**Reply by brianlangseth-NOAA** [reply]
but if I wanted to compare to an linf estimate, how would I do that. L2 in the model is at Linf, so Id like to compare it with an Linf estimate
`999 #_Growth_Age_for_L2 (999 to use as Linf)`

**Reply by okenk** [reply]
Ah. My math says as a2 -> infinity, the denominator -> 1 and can be eliminated. (Denominator is `(1-exp(-k[Sex]*24))`)

**Reply by brianlangseth-NOAA** [reply]
So then I could rerun that script without the `/(1 - exp(-k[Sex] * 24))` part to get new estimates corresponding to Linf?

**Reply by okenk** [reply]
I believe so!

**Reply by brianlangseth-NOAA** [reply]
I am struggling pulling the data. Using [these lines](https://github.com/okenk/Assessment_Class/blob/c50f538b69edf964d3928f3cd076d82bd000d4ed/code/W03_growth.R#L9-L13) throws an error, but this data has to be saved somewhere. Can I pull from [here](https://github.com/pfmc-assessments/canary_2023/blob/5890827095a9b02df969779a1ee3b1a119886352/code/canary_surveys_workup.R#L28) and rename so as to match the code from the class?

**Reply by okenk** [reply]
Yes, that should work.

**Reply by okenk** [reply]
As long as you only use wcgbts data. The triennial on gdrive doesn't have the 1986 otoliths yet

**Comment by brianlangseth-NOAA** [comment]
Haha. Finally got it. The google drive [file locator](https://github.com/pfmc-assessments/canary_2023/blob/5890827095a9b02df969779a1ee3b1a119886352/code/canary_surveys_workup.R#L28) must be unique to you because it didn't work for me. I went to my drive and copied the one there and it worked. 

```
Formula: Length_cm ~ la1[Sex] + (la2[Sex] - la1[Sex]) * (1 - exp(-k[Sex] * 
    (Age - 1)))

Parameters:
      Estimate Std. Error t value Pr(>|t|)    
la11 11.382156   0.164262   69.29   <2e-16 ***
la12 11.365088   0.147812   76.89   <2e-16 ***
la21 57.900273   0.188344  307.42   <2e-16 ***
la22 51.320627   0.119491  429.49   <2e-16 ***
k1    0.142700   0.001589   89.79   <2e-16 ***
k2    0.175495   0.001670  105.10   <2e-16 ***

```

**Comment by brianlangseth-NOAA** [comment]
Here is some code to compare the external and internal growth estimates. Couldn't figure how to plot over the plot from SSplots without spending more time on it
```
Length_cm ~ la1[Sex] + (la2[Sex] - la1[Sex]) * (1 - exp(-k[Sex] * 
                                                          (Age - 1)))

age = c(1:40)
lmm <- 8.344 + (54.6481000	 - 8.344) * (1-exp(-0.1558210 * (age-1)))
lfm <- 8.344 + (59.4463000 - 8.344) * (1-exp(-0.1372000	 * (age-1)))
lmx <- 11.365088 + (51.320627 - 11.365088) * (1-exp(-0.175495  * (age-1)))
lfx <- 11.382156  + (57.900273	 - 11.382156 ) * (1-exp(-0.142700 * (age-1)))


plot(age,lfm,type="l",col=2)
lines(age,lmm,col=4)
lines(age,lfx,col=2,lty=2)
lines(age,lmx,col=4,lty=2)
```
